crates/ruff_python_parser/tests/snapshots/invalid_syntax@with_items_parenthesized_missing_comma.py.snap

```diff
@@ -239,42 +239,49 @@ Module(
             ),
             With(
                 StmtWith {
-                    range: 136..160,
+                    range: 136..159,
                     is_async: false,
                     items: [
                         WithItem {
-                            range: 142..147,
-                            context_expr: Name(
-                                ExprName {
-                                    range: 142..147,
-                                    id: "item1",
-                                    ctx: Load,
-                                },
-                            ),
-                            optional_vars: None,
-                        },
-                        WithItem {
-                            range: 149..154,
-                            context_expr: Name(
-                                ExprName {
-                                    range: 149..154,
-                                    id: "item2",
+                            range: 141..154,
```

Comment:
The old parse tree here was probably slightly better.

Reply:
Yeah, missing closing parenthesis is difficult to recover from because it's occurring in list parsing which skips over the `:` token.

Reply:
Should the recovery skip over the `:`? Shouldn't it stop when seeing a `:` because it is part of the with items recovery set?

Reply:
For the initial assumption of parenthesized with-items, the `:` is not part of the recovery set (maybe it should?) and so we don't stop at `:` which means we can't reliably detect whether our assumption was correct. This is why it falls back to parenthesized expression and parsing it as a tuple expression.

Reply:
Yeah, i think it probably should. Or we risk running over the end of the case header

Reply:
Done in https://github.com/astral-sh/ruff/pull/11775